cpp/src/io/parquet/page_enc.cu

```diff
@@ -1315,56 +1315,44 @@ __global__ void __launch_bounds__(128, 8)
   }
 }
 
-// blockDim(128, 1, 1)
-__global__ void __launch_bounds__(128) gpuDecideCompression(device_span<EncColumnChunk> chunks)
+// blockDim(32, 1, 1)
+__global__ void __launch_bounds__(32) gpuDecideCompression(device_span<EncColumnChunk> chunks)
 {
-  // After changing the way structs are loaded from coop to normal, this kernel has no business
-  // being launched with 128 thread block. It can easily be a single warp.
   __shared__ __align__(8) EncColumnChunk ck_g;
-  __shared__ __align__(4) unsigned int error_count;
+  __shared__ __align__(4) unsigned int compression_error;
   using warp_reduce = cub::WarpReduce<uint32_t>;
   __shared__ typename warp_reduce::TempStorage temp_storage[2];
-  __shared__ volatile bool has_compression;
-
-  uint32_t t                      = threadIdx.x;
-  uint32_t uncompressed_data_size = 0;
-  uint32_t compressed_data_size   = 0;
-  uint32_t num_pages;
 
+  auto const t = threadIdx.x;
   if (t == 0) {
-    ck_g = chunks[blockIdx.x];
-    atomicAnd(&error_count, 0);
-    has_compression = false;
+    ck_g              = chunks[blockIdx.x];
+    compression_error = 0;
   }
-  __syncthreads();
-  if (t < 32) {
-    num_pages = ck_g.num_pages;
-    for (uint32_t page = t; page < num_pages; page += 32) {
-      auto& curr_page         = ck_g.pages[page];
-      uint32_t page_data_size = curr_page.max_data_size;
-      uncompressed_data_size += page_data_size;
-      if (auto comp_res = curr_page.comp_res; comp_res != nullptr) {
-        has_compression = true;
-        compressed_data_size += comp_res->bytes_written;
-        if (comp_res->status != compression_status::SUCCESS) { atomicAdd(&error_count, 1); }
-      }
+  __syncwarp();
+
+  uint32_t uncompressed_data_size = 0;
+  uint32_t compressed_data_size   = 0;
+  auto const num_pages            = ck_g.num_pages;
+  for (auto page = t; page < num_pages; page += warpSize) {
```

Comment:
We don't use `warpSize` in libcudf kernels. We always use `cudf::detail::warp_size` instead. https://github.com/rapidsai/cudf/blob/52342789a367153bedbc804efdc0a54a7b6ed083/cpp/include/cudf/detail/utilities/cuda.cuh#L36

This is important for various reasons -- we cannot assume that hardware will have a constant warp size of 32 forever. While we're not in a particularly good place to refactor the code for non-32 warp sizes, this would at least help us track where warp size assumptions are being used.

This rule is violated in [one other place, in merge.cu](https://github.com/rapidsai/cudf/blob/52342789a367153bedbc804efdc0a54a7b6ed083/cpp/src/merge/merge.cu#L100) which we should fix.

See also:
- https://github.com/rapidsai/cudf/pull/11869#discussion_r998705107
- https://github.com/rapidsai/cudf/pull/3842#discussion_r373320116